var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -44,16 +45,19 @@ class Mercury(CMakePackage):
             description='Checksum verify all request/response messages')
 
     depends_on('cmake@2.8.12.2:', type='build')
-    # depends_on('cci', when='+cci')  # TODO: add CCI package
     depends_on('bmi', when='+bmi')
     depends_on('mpi', when='+mpi')
-    depends_on('libfabric@1.5:', when='+ofi')
-    depends_on('openpa@1.0.3:', when='%gcc@:4.8')
+    depends_on('libfabric@1.5:', when='@:2.0.1+ofi')
+    depends_on('libfabric@1.7:', when='@2.1.0:+ofi')
```

Comment:
Here you can group directives for better readability:
```suggestion
    with when('+ofi'):
        depends_on('libfabric@1.5:', when='@:2.0.1')
        depends_on('libfabric@1.7:', when='@2.1.0:')
```